packages/api/lib/deleter.js

```diff
@@ -0,0 +1,47 @@
+
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+// class to receive POSTS with build information
+
+const firebaseEncode = require('../lib/firebase-encode');
+
+async function deleteRepo (client, repoId) {
+  const executeMe = [];
+
+  const allBuilds = await client.collection(global.headCollection).doc(firebaseEncode(repoId)).collection('builds').get();
```

Comment:
In the case of deleting a repository, we might also want the concept of a "tombstone" which indicates that the repository should 404, i.e., appear if it doesn't exist, but would prevent future uploads to the namespace.

I'm thinking of two scenarios:

* someone might upload an offensive repository name "as a joke" we want to block this behavior.
* someone might upload a repository name that gets flagged by automated [DMCA](https://www.eff.org/issues/dmca) detection.

Both these scenarios cropped up more than I would like at npm -- this is another reason not to have organization search as a feature, as it would surface offensive organization names in the typeahead search.

CC: @gis727 for context.

Reply:
I will add this feature to this repository, and keep the repository with only one field such as "flagged" and if so not allow new posts

Reply:
@StuartRucker I think we'll want both types of deleting, so we could just make this follow up work with a tracking issue.